src/engine/Server.cpp

```diff
@@ -338,8 +337,20 @@ Awaitable<void> Server::process(
       throw std::runtime_error(
           "Parameter \"query\" must not have an empty value");
     }
+    std::chrono::seconds timeLimit =
+        RuntimeParameters().get<"default-query-timeout">();
+    if (auto timeoutParam = checkParameter("timeout", std::nullopt)) {
+      std::string userTimeoutString{timeoutParam.value()};
+      // std::stoi does not support std::string_view
+      std::chrono::seconds userTimeout{std::stoi(userTimeoutString)};
```

Comment:
I think the query parameter should have the exact same format as the comman-line parameter etc, so you should also call your parsing function that requires `13s` as the format here.